core/subsurface-qt/DiveObjectHelper.cpp

```diff
@@ -405,15 +405,23 @@ QStringList DiveObjectHelper::getCylinder() const
 	return getCylinder;
 }
 
-QString DiveObjectHelper::startPressure() const
+QStringList DiveObjectHelper::startPressure() const
 {
-	QString startPressure = getPressures(m_dive, START_PRESSURE);
+	QStringList startPressure;
+	for (int i = 0; i < MAX_CYLINDERS; i++) {
+		if (is_cylinder_used(m_dive, i))
+			startPressure << getPressures(m_dive, i, START_PRESSURE);
+	}
 	return startPressure;
 }
 
-QString DiveObjectHelper::endPressure() const
+QStringList DiveObjectHelper::endPressure() const
 {
-	QString endPressure = getPressures(m_dive, END_PRESSURE);
+	QStringList endPressure;
+	for (int i = 0; i < MAX_CYLINDERS; i++) {
+		if (is_cylinder_used(m_dive, i))
+			endPressure << getPressures(m_dive, i, END_PRESSURE);
+	}
 	return endPressure;
 }
```

Comment:
DiveObjectHelper provides cylinderObjects(), returning a list of all individual cylinders with their info. 

Is there a reason not to use that instead of recreating the info as separate QStringList?

Reply:
Because we only want the cylinders that are used in a particular dive, not all cylinders specified in subsurface. Also it is only the cylinder name that is important at this stage.